# NGC 346 stars from Sabbi et al (2007)

In my never-ending quest to get a catalog of all the stars in NGC 346, I am now going to try using the astroquery Vizier interface to get the tables from: 

```bibtex
@article{Sabbi:2007h,
	author = {{Sabbi}, E. and {Sirianni}, M. and {Nota}, A. and {Tosi}, M. and {Gallagher}, J. and {Meixner}, M. and {Oey}, M.~S. and {Walterbos}, R. and {Pasquali}, A. and {Smith}, L.~J. and {Angeretti}, L.},
	journal = {\aj},
	month = jan,
	number = {1},
	pages = {44-57},
	title = {{Past and Present Star Formation in the SMC: NGC 346 and its Neighborhood}},
	volume = {133},
	year = 2007}

```

## Find which catalog we want

I had to install the `astroquery` library from conda-forge.

In [2]:
from astroquery.vizier import Vizier

I am following a similar method to the examples in the [astroquery docs](https://astroquery.readthedocs.io/en/latest/vizier/vizier.html):

In [6]:
catalog_list = Vizier.find_catalogs('Sabbi 2007')

We now have a list of matching catalogs:

In [5]:
for k,v in catalog_list.items():
    print(k)
    print(v.description, end="\n\n")

J/ApJ/730/78
Hubble and Spitzer photometry of NGC 602 (Carlson+, 2011)

J/ApJ/872/199
Candidate spectrophotometric standard DA WDs (Calamida+, 2019)

J/ApJS/222/11
Hubble Tarantula Treasury Project (HTTP). III. (Sabbi+, 2016)

J/ApJS/235/23
LEGUS galaxies1 observations (Sabbi+, 2018)

J/A+A/464/939
Abundances of red giants in NGC 6218 (Carretta+, 2007)

J/A+A/505/117
Abundances of red giants in 15 globular clusters (Carretta+, 2009)

J/A+A/598/A5
Gaia-ESO Survey iDR4 calibrators (Pancino+, 2017)

J/A+A/636/A54
Massive young stellar objects in 30 Doradus (van Gelder+, 2020)

J/AJ/133/44
NGC 346 HST VI photometry (Sabbi+, 2007)

J/AJ/136/1703
Ridgelines of six intermediate-age SMC clusters (Glatt+, 2008)

J/AJ/156/98
Runaway stars in the 30 Doradus region of the LMC (Platais+, 2018)

J/AJ/158/201
Young star cluster Westerlund 2 observed with MUSE (Zeidler+, 2019)

J/MNRAS/452/3508
Young stellar structures in NGC 6503 (Gouliermis+, 2015)



This gave more than I was expecting, but the one we want is `J/AJ/133/44`:

In [13]:
catalog_id = "J/AJ/133/44"

## Download the tables

We have to set `Vizier.ROW_LIMIT = -1` to make sure we get the complete table.  Otherwise, the default is to return only 50 rows.

In [12]:
Vizier.ROW_LIMIT = -1
catalogs = Vizier.get_catalogs(catalog_id)
catalogs

TableList with 2 tables:
	'0:J/AJ/133/44/table2' with 8 column(s) and 79960 row(s) 
	'1:J/AJ/133/44/table3' with 8 column(s) and 16 row(s) 

This gives two tables.  The first is a list of all 79960 stars that they have found in NGC 346.  The second is a list of the 16 sub-clusters that they have identified:

In [15]:
stars, subclusters = catalogs
subclusters

<Table length=16>
__SSN2007_  RAJ2000    DEJ2000    Rad    Age   e_Age   Npms   Dpms 
            "h:m:s"    "d:m:s"     pc    Myr    Myr         1 / pc2
   str5      str10       str9   float32 int16 float32 int16 float32
---------- ---------- --------- ------- ----- ------- ----- -------
     Sc  1 00 59 05.2 -72 10 28     2.5     3     1.0   342    17.4
     Sc  2 00 59 01.8 -72 10 35     2.3     3     1.0   295    17.8
     Sc  3 00 59 06.0 -72 10 43     1.6     3     1.0   138    17.2
     Sc  4 00 59 02.5 -72 10 07     0.6     3     1.0    20    17.7
     Sc  5 00 59 00.3 -72 10 04     1.1     3     1.0    32     8.4
     Sc  6 00 58 57.4 -72 09 55     1.2     3     1.0    14     3.1
     Sc  7 00 59 05.4 -72 10 57     1.5     3     1.0    65     9.2
     Sc  8 00 59 07.7 -72 10 48     1.9     3     1.0   109     9.6
     Sc  9 00 59 12.5 -72 11 08     1.0     3     1.0    29     9.2
     Sc 10 00 59 14.9 -72 11 01     1.6     3     1.0    61     7.6
     Sc 11 00 59 15.3 -72 11 12     1.1     3     1.0    38    10.0
     Sc 12 00 59 19.8 -72 11 20     1.5     3     1.0    25     3.5
     Sc 13 00 59 11.3 -72 10 00     1.9     3     1.0    90     7.9
     Sc 14 00 59 13.9 -72 09 27     0.9     3     1.0    22     8.6
     Sc 15 00 59 17.0 -72 09 15     0.6     3     1.0    21    18.6
     Sc 16 00 59 30.8 -72 09 10     1.6    15     2.5    65     8.0

## Look at the star catalog

In [16]:
stars

<Table length=79960>
  ID   F555W  e_F555W  F814W  e_F814W   RAJ2000      DEJ2000     MPG 
        mag     mag     mag     mag     "h:m:s"      "d:m:s"         
int32 float32 float32 float32 float32    str12        str12     int16
----- ------- ------- ------- ------- ------------ ------------ -----
    1  11.570   0.001  11.557   0.001 00 59 26.581 -72 09 53.00    --
    2  11.916   0.001  11.841   0.001 00 59 26.571 -72 09 53.99    --
    4  12.326   0.001  11.793   0.001 00 58 42.424 -72 09 43.27   185
    5  12.551   0.002  12.397   0.001 00 59 31.975 -72 10 46.11   789
    6  12.580   0.001  12.304   0.001 00 59 26.542 -72 09 53.00    --
    7  12.609   0.002  12.742   0.002 00 59 04.479 -72 10 24.77   435
    8  12.734   0.002  12.398   0.001 00 59 31.975 -72 10 46.13    --
    9  13.454   0.002  13.679   0.003 00 59 00.743 -72 10 28.16   355
   10  13.517   0.002  11.966   0.001 00 58 53.935 -72 12 04.78   283
  ...     ...     ...     ...     ...          ...          ...   ...
29426  26.820   0.077  25.924   0.081 00 58 34.453 -72 11 40.57    --
29495  26.842   0.098  25.283   0.050 00 59 31.898 -72 09 48.61    --
29510  26.830   0.081  25.513   0.049 00 59 21.847 -72 08 21.56    --
30841  26.864   0.085  25.586   0.050 00 58 44.239 -72 12 13.53    --
30948  26.874   0.091  25.156   0.044 00 58 44.130 -72 10 57.57    --
31171  26.885   0.089  25.437   0.052 00 58 45.559 -72 11 42.16    --
31212  26.880   0.095  25.328   0.054 00 58 31.282 -72 12 21.38    --
31969  26.895   0.098  25.172   0.048 00 59 29.197 -72 10 37.82    --
32517  26.917   0.093  25.508   0.059 00 58 44.016 -72 12 20.65    --
32733  26.928   0.086  25.444   0.051 00 59 36.172 -72 08 39.77    --

We want to convert the coordinate columns into degrees for ease of plotting, so we need some astropy libraries:

In [20]:
from astropy.coordinates import SkyCoord, Angle, Latitude, Longitude
import astropy.units as u

I had originally thought of using SkyCoord, but this is not necessary

In [26]:
# SkyCoord(stars["RAJ2000"], stars["DEJ2000"], unit=(u.hourangle, u.deg))

Easier to just convert the RA and DEC columns separately (note, these cells take a while to run – 10s of seconds on my laptop):

In [21]:
ra = Longitude(stars["RAJ2000"], unit=u.hourangle)

In [22]:
dec = Latitude(stars["DEJ2000"], unit=u.deg)

Check that the values in degrees look reasonable:

In [29]:
ra.to(u.deg), dec

(<Longitude [14.86075417, 14.8607125 , 14.67676667, ..., 14.87165417,
             14.6834    , 14.90071667] deg>,
 <Latitude [-72.16472222, -72.16499722, -72.16201944, ..., -72.17717222,
            -72.20573611, -72.14438056] deg>)

Add them in as new columns to the table and set display precision to about 10 mas. 

In [30]:
stars["ra"] = ra.to(u.deg)
stars["dec"] = dec
stars["ra"].format = "{:.6f}"
stars["dec"].format = "{:.6f}"
stars

<Table length=79960>
  ID   F555W  e_F555W  F814W  ...   DEJ2000     MPG      ra       dec    
        mag     mag     mag   ...   "d:m:s"             deg       deg    
int32 float32 float32 float32 ...    str12     int16  float64   float64  
----- ------- ------- ------- ... ------------ ----- --------- ----------
    1  11.570   0.001  11.557 ... -72 09 53.00    -- 14.860754 -72.164722
    2  11.916   0.001  11.841 ... -72 09 53.99    -- 14.860712 -72.164997
    4  12.326   0.001  11.793 ... -72 09 43.27   185 14.676767 -72.162019
    5  12.551   0.002  12.397 ... -72 10 46.11   789 14.883229 -72.179475
    6  12.580   0.001  12.304 ... -72 09 53.00    -- 14.860592 -72.164722
    7  12.609   0.002  12.742 ... -72 10 24.77   435 14.768662 -72.173547
    8  12.734   0.002  12.398 ... -72 10 46.13    -- 14.883229 -72.179481
    9  13.454   0.002  13.679 ... -72 10 28.16   355 14.753096 -72.174489
   10  13.517   0.002  11.966 ... -72 12 04.78   283 14.724729 -72.201328
  ...     ...     ...     ... ...          ...   ...       ...        ...
29426  26.820   0.077  25.924 ... -72 11 40.57    -- 14.643554 -72.194603
29495  26.842   0.098  25.283 ... -72 09 48.61    -- 14.882908 -72.163503
29510  26.830   0.081  25.513 ... -72 08 21.56    -- 14.841029 -72.139322
30841  26.864   0.085  25.586 ... -72 12 13.53    -- 14.684329 -72.203758
30948  26.874   0.091  25.156 ... -72 10 57.57    -- 14.683875 -72.182658
31171  26.885   0.089  25.437 ... -72 11 42.16    -- 14.689829 -72.195044
31212  26.880   0.095  25.328 ... -72 12 21.38    -- 14.630342 -72.205939
31969  26.895   0.098  25.172 ... -72 10 37.82    -- 14.871654 -72.177172
32517  26.917   0.093  25.508 ... -72 12 20.65    -- 14.683400 -72.205736
32733  26.928   0.086  25.444 ... -72 08 39.77    -- 14.900717 -72.144381